python/lsst/afw/multiband.py

```diff
@@ -0,0 +1,463 @@
+        functionality.
+
+        Parameters
+        ----------
+        filters: list
+            List of filter names.
+        singles: list
+            List of single band objects
+        """
+        self._filters = tuple([f for f in filters])
+        self._singles = tuple(singles)
+        self._bbox = self._singles[0].getBBox()
+        self._singleType = type(self._singles[0])
+
+        if not all([s.getBBox() == self.getBBox() for s in self.singles]):
+            bboxes = [s.getBBox() == self.getBBox() for s in self.singles]
+            err = "`singles` are required to have the same bounding box, received {0}"
+            raise ValueError(err.format(bboxes))
+        if not all([type(s) == self.singleType for s in self.singles]):
+            singleTypes = [type(s) == self.singleType for s in self.singles]
+            err = "`singles` are required to have the same type, received {0}"
+            raise ValueError(err.format(singleTypes))
+
+    def copy(self, deep=False):
+        """Copy the current object
+
+        This must be overloaded in a subclass of `MultibandBase`
+
+        Parameters
+        ----------
+        deep: bool
+            Whether or not to make a deep copy
+
+        Returns
+        -------
+        result: `MultibandBase`
+            copy of the instance that inherits from `MultibandBase`
+        """
+        err = "_copySingles must be implemented in an inherited class to enable copies"
+        raise NotImplementedError(err)
+
+    @property
+    def filters(self):
+        """List of filter names for the single band objects
+        """
+        return self._filters
+
+    @property
+    def singles(self):
+        """List of afw single band objects
+        """
+        return self._singles
+
+    @property
+    def singleType(self):
+        """Type of single band objects
+        """
+        return self._singleType
+
+    def getBBox(self):
+        """Bounding box
+        """
+        return self._bbox
+
+    def getXY0(self):
+        """Minimum coordinate in the bounding box
+        """
+        return self.getBBox().getMin()
+
+    def setXY0(self, xy0):
+        """Update the XY0 position for each single band object
+        """
+        bbox = self.getBBox()
+        self._bbox = Box2I(xy0, bbox.getDimensions())
+        for n in range(len(self)):
+            self.singles[n].setXY0(xy0)
+
+    @property
+    def x0(self):
+        """X0
+
+        X component of XY0 `Point2I.getX()`
+        """
+        return self.getBBox().getMinX()
+
+    @property
+    def y0(self):
+        """Y0
+
+        Y component of XY0 `Point2I.getY()`
+        """
+        return self.getBBox().getMinY()
+
+    @property
+    def origin(self):
+        """Minimum (y,x) position
+
+        This is the position of `self.getBBox().getMin()`,
+        but available as a tuple for numpy array indexing.
+        """
+        return np.array([self.y0, self.x0])
```

Comment:
Good point. I changed it hastily when making the docs so that `iy, ix = mImage.origin + np.array([2, 1])` works, but you're right that it should be a tuple and the user will just have to use `iy, ix = np.array(mImage.origin) + np.array([2, 1])`.